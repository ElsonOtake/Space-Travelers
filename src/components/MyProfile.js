import { useSelector } from 'react-redux';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import ListGroup from 'react-bootstrap/ListGroup';

const MyProfile = () => {
  const missionsData = useSelector((state) => state.missionsReducer);
  const rocketsData = useSelector((state) => state.rocketsReducer);

  return (
    <Container className="mt-4">
      <Row>
        <Col>
          <h4>My Missions</h4>
          <ListGroup>
            {
              missionsData.length > 0
              && missionsData.filter((mission) => mission.reserved).map((res) => (
                <ListGroup.Item key={res.id}>
                  {res.name}
                </ListGroup.Item>
              ))
            }
          </ListGroup>
        </Col>
        <Col>
          <h4>My Rockets</h4>
          <ListGroup>
            {
              rocketsData.length > 0
              && rocketsData.filter((rocket) => rocket.reserved).map((res) => (
                <ListGroup.Item key={res.id}>
                  {res.name}
                </ListGroup.Item>
              ))
            }
          </ListGroup>
        </Col>
      </Row>
    </Container>
  );
};

export default MyProfile;
