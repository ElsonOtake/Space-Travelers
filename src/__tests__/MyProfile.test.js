import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import MyProfile from '../components/MyProfile';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';

describe('Tests for the MyProfile component', () => {
  render(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  );
  test('Check for the following test on screen', () => {

    expect(screen.getByText('My Missions')).toBeInTheDocument();
    expect(screen.getByText('My Rockets')).toBeInTheDocument();

    expect(screen.queryByText('My Profile')).toBeNull();

    expect(screen.getAllByRole('heading').length).toBe(2);

});
  test('Check for the snapshot', () => {
    expect(screen.debug()).toMatchSnapshot();
  });
});