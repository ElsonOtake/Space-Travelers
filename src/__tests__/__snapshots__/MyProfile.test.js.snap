// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the MyProfile component Check for the snapshot 1`] = `undefined`;
